app/controllers/checkout_controller.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+# CheckoutController
+class CheckoutController < ApplicationController
+  include Wicked::Wizard
+  include CheckoutHelper
+  before_action :authenticate_user!, :set_order
+
+  steps :address, :delivery, :payment, :confirm, :complete
+
+  def show
+    redirect_to root_path if @order.books.count.zero?
+    case step
+    when :address then show_address
+    when :delivery then show_delivery
+    when :payment then show_payment
+    when :confirm  then show_confirm
+    when :complete then show_complete
+    end
+    render_wizard unless performed?
+  end
+
+  def update
+    case step
+    when :address then update_address
+    when :delivery then update_delivery
+    when :payment then update_payment
+    when :confirm  then update_confirm
+    when :complete then redirect_to root_path
+    end
+    redirect_to_valid_step
+  end
+
+  private
+
+  def show_address
+    @address_service = CheckoutAddressService.new(@order, nil, current_user)
```

Comment:
I think a form object will fit better here. It looks almost like your current service objects, but it will contain validation logic and will have a `persist` method instead of `call`.
Here are good examples:
https://www.sitepoint.com/7-design-patterns-to-refactor-mvc-components-in-rails/
https://robots.thoughtbot.com/activemodel-form-objects